packages/flutter/lib/src/widgets/debug.dart

```diff
@@ -22,3 +24,21 @@ bool debugCheckHasMediaQuery(BuildContext context) {
   });
   return true;
 }
+
+bool debugHasDuplicateKeys(Widget parent, Iterable<Widget> children) {
```

Comment:
I'd wrap the body of this function in an assert to make sure it is used only in debug builds.  See debugCheckHasMediaQuery above for the pattern.